pandas/core/dtypes/base.py

```diff
@@ -396,6 +396,16 @@ def _can_hold_na(self) -> bool:
         """
         return True
 
+    @property
+    def _is_immutable(self) -> bool:
+        """
+        Can arrays with this dtype be modified with __setitem__? If not, return
```

Comment:
Should `ExtensionArray.__setitem__` have a `if dtype._is_immutable: raise TypeError` check to solidify this?

Reply:
I guess that would be harmless since any working `EA.__setitem__` would need to override